lighthouse-core/audits/long-tasks.js

```diff
@@ -0,0 +1,97 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit.js');
+const NetworkRecords = require('../computed/network-records.js');
+const i18n = require('../lib/i18n/i18n.js');
+const MainThreadTasks = require('../computed/main-thread-tasks.js');
+const BootupTime = require('./bootup-time.js');
+
+const UIStrings = {
+  /** Title of a diagnostic LH audit that provides details on the longest running tasks that occur when the page loads. */
+  title: 'Avoid long main-thread tasks',
+  /** Description of a diagnostic LH audit that shows the user the longest running tasks that occur when the page loads. */
+  description: 'Lists the longest tasks on the main thread, ' +
+    'useful for identifying worst contributors to input delay. ' +
+    '[Learn more](https://web.dev/long-tasks-devtools)',
+  /** [ICU Syntax] Label identifying the number of long-running CPU tasks that occurred while loading a web page. */
+  displayValue: `{itemCount, plural,
+  =1 {# long task found}
+  other {# long tasks found}
+  }`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LongTasks extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'long-tasks',
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'devtoolsLogs'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const settings = context.settings || {};
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const tasks = await MainThreadTasks.request(trace, context);
+    const devtoolsLog = artifacts.devtoolsLogs[LongTasks.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+    const multiplier = settings.throttlingMethod === 'simulate' ?
+      settings.throttling.cpuSlowdownMultiplier : 1;
+
+    const jsURLs = BootupTime.getJavaScriptURLs(networkRecords);
+    // Only consider up to 20 long, top-level (no parent) tasks that have an explicit endTime
+    const longtasks = [...tasks]
+      .map(t => ({...t, duration: t.duration * multiplier}))
+      .filter(t => t.duration >= 50 && !t.unbounded && !t.parent)
+      .sort((a, b) => b.duration - a.duration)
+      .slice(0, 20);
+
+    const results = longtasks.map(task => ({
+      url: BootupTime.getAttributableURLForTask(task, jsURLs),
+      group: task.group.label,
+      start: task.startTime,
```

Comment:
I think it's going to be weird that `startTime` is reported on the actual timeline while `duration` is going to be on the simulated timeline, should we just exclude it for now?